### PR TITLE
Split steering standards by component and scope context loading

### PR DESCRIPTION
**Pull request by garretfick**
## Problem

The `specs/steering/` files had grown into overlapping grab-bags. `development-standards.md` was always loaded into AI context, yet roughly half of it was Rust/compiler-specific — so working on the VS Code extension or the docs website pulled in irrelevant guidance. Cross-cutting rules (git gate, problem-code templates, test naming) were also duplicated across three or more files, and three Kiro pointers were silently always-loaded because they were missing `fileMatch` frontmatter.

## Changes

**Component extraction (scope by path)**
- New `compiler-standards.md` (`compiler/**`) — module structure, testing, error handling, performance, Rust/`unsafe`/clippy rules.
- New `doc-standards.md` (`docs/**`) — renamed from the in-progress `documentation-standards.md`; content preserved.
- New `extension-standards.md` (`integrations/vscode/**`) — README sync + pointers to the extension testing gates and `E####` codes.
- Each gets a `fileMatch` Kiro pointer and a `CLAUDE.md` entry with an "especially relevant for `<path>`" hint (Claude's `fileMatch` equivalent).

**Slim the always-on core**
- `development-standards.md`: 447 → 298 lines, now cross-cutting process only (specs structure, planning, prefactoring, duplication principle, git workflow), plus a Component-Specific Standards index.

**Dedup**
- Full CI-pipeline explanation → single owner `common-tasks.md`.
- Problem-code CSV + `.rst` templates → single owner `problem-code-management.md`.

**Fix silently-always-loaded pointers**
- Added `fileMatch` frontmatter to `compiler-architecture.md`, `compatibility-library-authoring.md`, and `steering-file-guidelines.md`.

**Fix the root cause**
- Added a "Scope to the Narrowest Path" principle to `steering-file-guidelines.md`, clarified that "self-contained" means tool-agnostic (not restating other files' rules), and refreshed the stale file-tree example.

## Effect

The always-loaded Kiro set drops from 6 files to 3 (`development-standards`, `common-tasks`, `glossary`). Extension and docs sessions no longer load the compiler/Rust standards.

Documentation-only change (markdown steering files); no compiler CI gate applies. Cross-reference anchors were verified to resolve.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: specs/steering/compiler-standards.md [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+- **No global state dependencies** - each test must be self-contained
+- **Terminate on failure** - use `assert!`, `assert_eq!`, etc. rather than continuing
+- **One assertion per logical concept** - but multiple assertions for the same concept are fine
+- **No panic! in tests** - Use `assert!` macros instead of `panic!()` for test failures
+  - ❌ Bad: `match result { Ok(x) => assert_eq!(x, 5), _ => panic!("Expected Ok") }`
+  - ✅ Good: `assert!(result.is_ok()); assert_eq!(result.unwrap(), 5);`
+  - ✅ Better: `assert!(matches!(result, Ok(5)))`
+
+### Test Organization
+- Group related tests in the same module
```
Comment (garretfick) [inline specs/steering/compiler-standards.md:57]: Say use rstest to group tests and eliminate boiler plate code.

File: specs/steering/compiler-standards.md [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+- Include both positive and negative test cases
+- Test edge cases and boundary conditions
+
+For where each kind of test lives (VM, codegen, plc2plc, parser) and which leg
+asserts what, see [compiler-architecture.md](compiler-architecture.md#testing-architecture)
+and [syntax-support-guide.md](syntax-support-guide.md).
+
+## Error Handling
+
+Every user-facing error message **must** have a unique `P####` problem code. The
```
Comment (garretfick) [inline specs/steering/compiler-standards.md]: This needs to say a unique problem code. Problem codes for the compiler are `P####`; problem code for the runtime are `V####`.

File: specs/steering/compiler-standards.md [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+and the diagnostic-construction patterns — lives in
+[problem-code-management.md](problem-code-management.md). Do not restate that
+material here.
+
+When writing compiler code:
+- Prefer `Result<T, Diagnostic>` for fallible operations; propagate with `?`
+- Collect multiple diagnostics rather than failing on the first, where practical
+- Provide clear, actionable error messages: include what was expected vs. found
+- Use `Diagnostic::problem()` with an appropriate `Label::span()` for source location
+- Match IEC 61131-3 terminology in messages
```
Comment (garretfick) [inline specs/steering/compiler-standards.md:87]: Also mention to use `internal_error `and `not_supported` as special cases.

File: specs/steering/compiler-standards.md [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+### Compilation Speed
+- Optimize for reasonable compilation times
+- Use efficient algorithms for type checking and semantic analysis
+- Profile performance-critical paths when needed
+
+## Code Quality
+
+### Rust Best Practices
+- Do not suppress clippy warnings with `#[allow(...)]` — fix the underlying code instead. The only acceptable exception is `#[allow(dead_code)]` or `#[allow(unused_*)]` for in-progress code that is not yet wired up; remove these suppressions once the code is complete
+- Prefer `Result<T, E>` for error handling over panics
```
Comment (garretfick) [inline specs/steering/compiler-standards.md]: Never use panic or similar.

File: specs/steering/compiler-standards.md [hunk 5]
```diff
@@ -0,0 +1,134 @@
...
+- Use appropriate visibility modifiers (`pub`, `pub(crate)`, etc.)
+- Follow Rust naming conventions and idioms
+
+### Safety
+- Leverage Rust's safety guarantees
+- **`unsafe` code is rejected at compile time.** The workspace sets `unsafe_code = "deny"` in `[workspace.lints.rust]` (root `compiler/Cargo.toml`), and every member crate inherits it via `[lints] workspace = true`. Any `unsafe` block, function, trait, or impl in IronPLC code fails the build
+- **Do not bypass the check with `#[allow(unsafe_code)]`.** The standards already forbid `#[allow(...)]` suppressions (see [Rust Best Practices](#rust-best-practices)); `unsafe_code` is no exception. `deny` (rather than `forbid`) is the chosen level only so that proc-macros which wrap unsafe internally — e.g. `ctor::ctor`, whose expansion includes `#[allow(unsafe_code)]` — keep working. If a feature appears to require `unsafe`, raise it for discussion
+- Use strong typing to prevent logic errors (e.g., `TypeName` vs `String`)
```
Comment (garretfick) [inline specs/steering/compiler-standards.md]: Also add example for renaming u16 and so on

File: specs/steering/development-standards.md [hunk 6]
```diff
@@ -99,25 +107,40 @@ The completeness half is weaker than it reads. A requirement counts as tested wh
...
+### Required Steps
+
+**Planning**
+
+1. AI researches the code base and desired changes
+2. AI creates a **plan branch**, writes the plan to `specs/plans/YYYY-MM-DD-short-description.md`, and creates a PR for the plan.
+3. A person reviews and provides feedback on the plan until the plan is approved. This PR is never merged.
 
-All non-trivial features and changes **must** begin with an implementation plan. The plan is a branch-local artifact: it is reviewed as a file diff, then deleted before merge.
+**Preparation**
```
Comment (garretfick) [inline specs/steering/development-standards.md]: Let's change the "Preparation" title to "Prefactoring"

File: specs/steering/steering-file-guidelines.md [hunk 7]
```diff
@@ -424,37 +441,35 @@ For AI-assisted development patterns, see `specs/steering/`.
...
-├── settings.json                        # SessionStart hook (installs just)
-└── commands/                            # Slash commands (skill pointers)
-    ├── build.md                         # /project:build
-    ├── test.md                          # /project:test
-    ├── ci.md                            # /project:ci
-    └── format.md                        # /project:format
-
-CLAUDE.md                                # Claude entry point
-                                         # Steering docs + skills list
+specs/steering/                        # Single source of truth (full docs)
```
Comment (garretfick) [inline specs/steering/steering-file-guidelines.md]: Do we really need this table? It seems this would easily get out of date.

**Review by garretfick — COMMENTED** [review]
See inline comments
